pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -118,11 +121,70 @@ func (pl *VolumeBinding) EventsToRegister() []framework.ClusterEventWithHint {
 		// When CSIStorageCapacity is enabled, pods may become schedulable
 		// on CSI driver & storage capacity changes.
 		{Event: framework.ClusterEvent{Resource: framework.CSIDriver, ActionType: framework.Add | framework.Update}},
-		{Event: framework.ClusterEvent{Resource: framework.CSIStorageCapacity, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.CSIStorageCapacity, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterCSIStorageCapacityChange},
 	}
 	return events
 }
 
+func (pl *VolumeBinding) isSchedulableAfterCSIStorageCapacityChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
```

Comment:
How often is `CSIStorageCapacity` updated in an actual world?

If it's much infrequent in general, can we just return Queue when CSIStorageCapacity is created/updated and a Pod has specific .spec.volumes?

e.g., 

```go
func (pl *VolumeBinding) isSchedulableAfterCSIStorageCapacityChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
//...
	for _, vol := range pod.Spec.Volumes {
		if vol.PersistentVolumeClaim != nil || vol.Ephemeral != nil:
			// This Pod might have got unschedulable due to CSIStorageCapacity in a past scheduling cycle.			
			return framework.Queue, nil 
		}
	}
	return framework.QueueSkip, nil
```

Reply:
> How often is CSIStorageCapacity updated in an actual world?

CSIStorageCapacity is updated by the external-provisioner of each CSI Driver. The relevant code can be found here:
https://github.com/kubernetes-csi/external-provisioner/blob/v5.0.0/pkg/capacity/capacity.go#L592-L669

There are several events that trigger the update, but with the default settings, each CSIStorageCapacity for a single CSI Driver is updated at least every minute:
https://github.com/kubernetes-csi/external-provisioner/blob/v5.0.0/pkg/capacity/capacity.go#L512-L526
https://github.com/kubernetes-csi/external-provisioner/blob/v5.0.0/cmd/csi-provisioner/csi-provisioner.go#L111

The number of CSIStorageCapacity objects depends on the NodeTopology configuration of the target CSI Driver.
For example, when using network volumes like the gcp-compute-persistent-disk-csi-driver, it can be on a per-zone basis:
https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/v999.1.0/pkg/gce-pd-csi-driver/node.go#L377-L392
When using lvm running on each node, like topolvm, CSIStorageCapacity is created on a per-node basis:
https://github.com/topolvm/topolvm/blob/main/internal/driver/node.go#L557-L566

> If it's infrequent in general, can we just return Queue when CSIStorageCapacity is created/updated and a Pod has specific .spec.volumes?

Yes, that approach alone can help skip queueing Pods unrelated to CSIStorageCapacity, reducing unnecessary overhead.